File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        vector<int> stack;
        string res="";

        for (char chr : s) {
            if (chr==')') stack.pop_back();
            if (!stack.empty()) res += chr;
            if (chr=='(') stack.push_back(chr);
        }
        return res;
    }
};

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.rb
# @param {String} s
# @return {String}
def remove_outer_parentheses(s)
    stack=[]
    res=''
    s.each_char do |i|
        stack.pop if s[i] == ')'
        res += s[i] unless stack.empty?
        stack.push('(') if s[i] == '('    
    end
    res
end

